src/main/scala/com/fulcrumgenomics/vcf/MakeMixtureVcf.scala

```diff
@@ -0,0 +1,268 @@
+  @arg(flag="N", doc="Treat no-calls for samples as hom-ref genotypes.") val noCallIsHomRef: Boolean = true,
+  @arg(flag="a", doc="Format field containing allele fraction.") val alleleFractionField: Option[String] = None,
+  @arg(flag="p", doc="Digits of precision in generated allele fractions.") val precision: Int = 5
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  /**
+    * If no sample names are provided creates Sample objects for all the samples in the VCF
+    * in equal proportion.  If sample names are provided, creates Sample objects for each,
+    * auto-sets the proportion for any that do not have it specified, then validates that
+    * all specified samples exist in the VCF.
+    *
+    * @param header the input VCF header
+    * @return the set of samples with proportions
+    */
+  private[vcf] def determineSamples(header: VCFHeader): Seq[Sample] = this.samples match {
+    case Seq() =>
+      val ss       = header.getSampleNamesInOrder
+      val fraction = 1 / ss.size().toDouble
+      header.getSampleNamesInOrder.map(s => Sample(s, fraction)).toSeq
+    case sampleStrings =>
+      // Extract the samples and fractions from their strings
+      var samples = sampleStrings.map { s =>
+        s.lastIndexOf('@') match {
+          case -1 => Sample(s)
+          case  i => Sample(s.substring(0, i), s.substring(i+1).toDouble)
+        }
+      }
+
+      // Validate that each sample was specified only once
+      validate(samples.size == samples.map(_.name).distinct.size, "Each sample name can be specified at most once.")
+
+      // Auto-calculate fraction for any samples that didn't have it
+      val total = samples.map(_.proportion).sum
+      validate(total <= 1, "Sample fractions added to more than 1.0")
+      val remainder = 1 - total
+      val fraction  = remainder / samples.count(_.proportion == 0)
+      samples = samples.map(s => if (s.proportion == 0) s.copy(proportion = fraction) else s)
+      validate(samples.map(_.proportion).sum == 1, "Sample fractions must add to 1.0")
+
+      // Then lastly validate that all given samples exist in the VCF
+      val missing = samples.filterNot(s => header.getSampleNamesInOrder.contains(s.name))
+      validate(missing.isEmpty, s"Samples are not present in the VCF: ${missing.mkString(", ")}")
+
+      samples
+  }
+
+  /** The main execute method that reads the inputs, mixes and writes the outputs. */
+  override def execute(): Unit = {
+    val in            = new VCFFileReader(input.toFile, false)
+    val samples       = determineSamples(in.getFileHeader)
+    val sampleNameSet = CollectionUtil.makeSet(samples.map(_.name):_*)
+    val out           = MakeMixtureVcf.makeWriter(output, in.getFileHeader, MakeMixtureVcf.HeaderLines, this.outputSampleName)
+    val fmt           = new DecimalFormat("0." + ("#" * precision))
+
+    logger.info("Making mixture of: ", samples.mkString(", "))
```

Comment:
I think the current prefix makes more sense when you realize the output looks like:
`Making mixture of: s1@0.1, s2@0.2, s3@.07`
Which I read as "Making mixture of s1 at 10%, s2 at 20% and s3 at 70%."